plugins/modules/foreman_search_facts.py

```diff
@@ -107,16 +123,18 @@ def main():
             resource=dict(type='str', required=True),
             search=dict(default=""),
             full_details=dict(type='bool', aliases=['info'], default='false'),
+            params=dict(type='dict', default=dict()),
```

Comment:
```suggestion
            params=dict(type='dict'),
```

no need for a `default`, it will be `None` if not defined and `list_resource` will handle that fine